source/extensions/filters/udp/udp_proxy/udp_proxy_filter.h

```diff
@@ -106,7 +114,7 @@ class UdpProxyFilterConfig {
 
   Upstream::ClusterManager& cluster_manager_;
   TimeSource& time_source_;
-  const std::string cluster_;
+  Router::RouterConstSharedPtr router_;
```

Comment:
It looks like the way this member is used, it can be an object instead of a shared pointer. Would that work?

Reply:
In fact, I do not think so. `Router` is abstract, so we cannot use its object here.